lib/Doctrine/Migrations/MigrationTableUpdater.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\Table;
+use function in_array;
+
+class MigrationTableUpdater
+{
+    /** @var Connection */
+    private $connection;
+
+    /** @var AbstractSchemaManager */
+    private $schemaManager;
+
+    /** @var MigrationTable */
+    private $migrationTable;
+
+    /** @var AbstractPlatform */
+    private $platform;
+
+    public function __construct(
+        Connection $connection,
+        AbstractSchemaManager $schemaManager,
+        MigrationTable $migrationTable,
+        AbstractPlatform $platform
+    ) {
+        $this->connection     = $connection;
+        $this->schemaManager  = $schemaManager;
+        $this->migrationTable = $migrationTable;
+        $this->platform       = $platform;
+    }
+
+    public function updateMigrationTable() : void
+    {
+        $fromTable = $this->getFromTable();
+        $toTable   = $this->migrationTable->getDBALTable();
+
+        $fromSchema = $this->createSchema([$fromTable]);
+        $toSchema   = $this->createSchema([$toTable]);
+
+        $queries = $fromSchema->getMigrateToSql($toSchema, $this->platform);
+
+        foreach ($queries as $query) {
```

Comment:
Shouldn't this actually run inside transaction sice there are multiple queries?

Reply:
From my experience, PostgreSQL definitely does, MySQL definitely does not.
More vendors documented here: https://wiki.postgresql.org/wiki/Transactional_DDL_in_PostgreSQL:_A_Competitive_Analysis

Reply:
I added the transaction, it won't hurt :+1: